integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/ItMiiSampleHelper.java

```diff
@@ -0,0 +1,376 @@
+import static oracle.weblogic.kubernetes.TestConstants.OKD;
+import static oracle.weblogic.kubernetes.TestConstants.RESULTS_ROOT;
+import static oracle.weblogic.kubernetes.TestConstants.WEBLOGIC_IMAGE_NAME;
+import static oracle.weblogic.kubernetes.TestConstants.WEBLOGIC_IMAGE_TAG;
+import static oracle.weblogic.kubernetes.assertions.TestAssertions.doesImageExist;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createOcirRepoSecret;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createSecretForBaseImages;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.dockerLoginAndPushImageToRegistry;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.installAndVerifyOperator;
+import static oracle.weblogic.kubernetes.utils.TestUtils.getDateAndTimeStamp;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Helper class to test and verify MII sample using or not-using auxiliary image.
+ */
+@DisplayName("Helper class to test model in image sample")
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@IntegrationTest
+public class ItMiiSampleHelper {
+
+  private static final String MII_SAMPLES_WORK_DIR = RESULTS_ROOT
+      + "/model-in-image-sample-work-dir";
+  private static final String MII_SAMPLES_SCRIPT =
+      "../operator/integration-tests/model-in-image/run-test.sh";
+
+  private static final String CURRENT_DATE_TIME = getDateAndTimeStamp();
+  private static final String MII_SAMPLE_WLS_IMAGE_NAME_V1 = DOMAIN_IMAGES_REPO + "mii-" + CURRENT_DATE_TIME + "-wlsv1";
+  private static final String MII_SAMPLE_WLS_IMAGE_NAME_V2 = DOMAIN_IMAGES_REPO + "mii-" + CURRENT_DATE_TIME + "-wlsv2";
+  private static final String MII_SAMPLE_JRF_IMAGE_NAME_V1 = DOMAIN_IMAGES_REPO + "mii-" + CURRENT_DATE_TIME + "-jrfv1";
+  private static final String MII_SAMPLE_JRF_IMAGE_NAME_V2 = DOMAIN_IMAGES_REPO + "mii-" + CURRENT_DATE_TIME + "-jrfv2";
+  private static final String SUCCESS_SEARCH_STRING = "Finished without errors";
+
+  private static String opNamespace = null;
+  private static String domainNamespace = null;
+  private static String traefikNamespace = null;
+  private static String dbNamespace = null;
+  private static Map<String, String> envMap = null;
+  private static LoggingFacade logger = null;
+  private static boolean previousTestSuccessful = true;
+  private static DomainType domainTypeName = null;
+  private static ImageType imageTypeName = null;
+
+  public enum DomainType {
+    JRF,
+    WLS
+  }
+
+  public enum ImageType {
+    MAIN,
+    AUX
+  }
+
+  /**
+   * Install Operator.
+   * @param namespaces list of namespaces created by the IntegrationTestWatcher by the
+   *        JUnit engine parameter resolution mechanism
+   */
+  public static void initAll(List<String> namespaces) {
```

Comment:
minor: initAll could take domainType and imageType as parameters  - so it becomes more like a constructor, and the methods 'setDomainType' and 'setImageType' could then be removed

Reply:
changed